src/test/java/org/jenkinsci/plugins/workflow/cps/CpsVmExecutorServiceTest.java

```diff
@@ -160,6 +160,24 @@
         }
     }
 
+    @Issue({"JENKINS-58620"})
+    @Test public void wrongCatcherGroovyShellEvaluate() throws Exception {
```

Comment:
This can just be added to `wrongCatcher` I think. (Or the class could be reworked a bit to use `JenkinsRule` as a `ClassRule`.)

Reply:
I don't really like to put everything in one place and unfortunately have not enough knowledge of the logic that is tested - so can't properly rewrite the class for optimization...

Reply:
Should probably be put on hold until a plugin maintainer can refactor this class accordingly.

Reply:
You mean we have to wait for some not planned action just to merge the unit tests? Looks not promising at all. If you really want - I can move this unit test to `wrongCatcher` if it will make any better to be close to merge. But I really think it's not necessary for the current case.

Reply:
Well we need to wait until upstream is `deploy`ed anyway.

Reply:
Sure. I updated a little bit - removed curved brackets since they don't needed in `@Issue`.

Reply:
Assuming #320 is merged, you should be able to add a new test without adding significantly to execution time.

Reply:
Ok, thank you - will check and rewrite accordingly.

Reply:
I will prepare a change when #320 will be merged - because most probably it will conflict with my changes.